src/Microsoft.OData.Edm/Schema/EdmAnnotationsTarget.cs

```diff
@@ -0,0 +1,55 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmAnnotationsTarget.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using Microsoft.OData.Edm.Vocabularies;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// Represents an EDM annotations target.
+    /// </summary>
+    public class EdmAnnotationsTarget : IEdmAnnotationsTarget
+    {
+        private IEnumerable<IEdmElement> targetSegments;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmAnnotationsTarget"/> class.
+        /// </summary>
+        /// <param name="targetSegments">Target segments.</param>
+        public EdmAnnotationsTarget(IEnumerable<IEdmElement> targetSegments)
+        {
+            EdmUtil.CheckArgumentNull(targetSegments, nameof(targetSegments));
+
+            // Validate that the first element is Container.
+            if (!(targetSegments.First() is IEdmEntityContainer))

```

Comment:
>First [](http://example.com/codeflow?start=33&length=5)

First() throws exception if the list/enumerable is empty. 
Please be aware of it and add clear validation.